client/block.c

```diff
@@ -70,13 +76,26 @@ struct block_backrefs {
 #define ourprev link[MAX_LINKS - 2]
 #define ournext link[MAX_LINKS - 1]
 
+struct cache_block {
+        struct ldus_rbtree node;
+	xdag_hash_t hash;
+        struct xdag_block block;
+        struct cache_block *next;
+};
+
+
 static xdag_amount_t g_balance = 0;
 static xdag_time_t time_limit = DEF_TIME_LIMIT, xdag_era = XDAG_MAIN_ERA;
-static struct ldus_rbtree *root = 0;
+static struct ldus_rbtree *root = 0, *cache_root = 0;
```

Comment:
Can you clarify please what is the purpose of the new tree?

Reply:
to binary search the block in the cache

Reply:
But what is the purpose of that cache?
We already have cached blocks in another structure (root)